aiida_mlip/data/config.py

```diff
@@ -0,0 +1,148 @@
+
+from aiida_mlip.helpers.converters import convert_to_nodes
+
+
+class JanusConfigfile(SinglefileData):
+    """
+    Define config file type in AiiDA in yaml.
+
+    Parameters
+    ----------
+    file : Union[str, Path]
+        Absolute path to the file.
+    filename : Optional[str], optional
+        Name to be used for the file (defaults to the name of provided file).
+
+    Attributes
+    ----------
+    filepath : str
+        Path of the mlip model.
+
+    Methods
+    -------
+    set_file(file, filename=None, architecture=None, **kwargs)
+        Set the file for the node.
+    local_file(file, architecture, filename=None):
+        Create a ModelData instance from a local file.
+    download(url, architecture, filename=None, cache_dir=None, force_download=False)
+        Download a file from a URL and save it as ModelData.
+
+    Other Parameters
+    ----------------
+    **kwargs : Any
+        Additional keyword arguments.
+    """
+
+    def __init__(
+        self,
+        file: Union[str, Path],
+        filename: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Initialize the ModelData object.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Absolute path to the file.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+        Other Parameters
+        ----------------
+        **kwargs : Any
+            Additional keyword arguments.
+        """
+        super().__init__(file, filename, **kwargs)
+        self.base.attributes.set("filepath", str(file))
+
+    def set_file(
+        self,
+        file: Union[str, Path],
+        filename: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Set the file for the node.
+
+        Parameters
+        ----------
+        file : Union[str, Path]
+            Absolute path to the file.
+        filename : Optional[str], optional
+            Name to be used for the file (defaults to the name of provided file).
+
+        Other Parameters
+        ----------------
+        **kwargs : Any
+            Additional keyword arguments.
+        """
+        super().set_file(file, filename, **kwargs)
+        self.base.attributes.set("filepath", str(file))
+
+    def read_yaml(self) -> dict:
+        """
+        Convert yaml file to dictionary.
+
+        Returns
+        -------
+        dict
+            Returns the converted dictionary with the stored parameters.
+        """
+        with open(self.filepath, encoding="utf-8") as handle:
+            return yaml.safe_load(handle)
+
+    def store_content(self, store_all: bool = False, skip: list = None) -> dict:
+        """
+        Store the content of the config file in the database.
+
+        Parameters
+        ----------
+        store_all : bool
+            Define if you want to store all the parameters or only the main ones.
```

Comment:
good point.
at the moment I am saving:
code, structure, model, architecture, ensemble and fully_opt, I can add others if we want.
this means that when you run aiida-mlip using the config file, only this parameters will be saved in the data provenance as single parameters. still all the other info in the config file are saved in the config file node, so they are not lost, they are just grouped under configfile basically.

Reply:
the fact that there is the option to choose to store all, is not a real option tbh. I put it there for the implementation of the config for aiida itself, but now that I removed it I could probably remove also this option and keep it simple

Reply:
ways this could be improved:
-give people a choice on which parameters they want to save singularly (I don't think we should, tho, it gives space to too many errors)
-explain in the documentation or somewhere that when using the config file only some parameters are saved individually, the rest can be found in config (which in the end is the same as passing a list of parameters as a dictionary so I think it's fine)

Reply:
Yeah I think we probably want a set way that things are saved, outside of user control. There's probably not a particularly nice way to do anything else anyway.

So yes, I think the main thing is just making it clear (in the Python and the docs) what is saved differently